rasterio/_base.pyx

```diff
@@ -108,11 +108,24 @@ def driver_can_create_copy(drivername):
 cdef class DatasetBase(object):
     """Dataset base class."""
 
-    def __init__(self, path, options=None):
+    def __init__(self, path=None, options=None):
```

Comment:
@geowurster it's an interesting idea, I'll have to ponder it. Let's discuss it in a different ticket.